lib/un_locode/locode.rb

```diff
@@ -29,9 +29,9 @@ def self.find_by_name_and_function name, function, limit = 10
     end
 
     def self.find_by_locode(locode)
-      locode = locode.split(' ')
-      includes(:country).where(city_code: locode.last)
-        .where(countries: { code: locode.first }).first
+      locode = locode.gsub(' ','').upcase
+      includes(:country).where(city_code: locode[2..4])
+        .where(countries: { code: locode[0..1] }).first
```

Comment:
maybe you could use `find_by` matchers instead of `where`. Then you can skip first, also when looking up a locode it should only match 1 entry